### ISSUE TITLE
Segmentation fault in QXcbScreen::mapFromNative in 0.12.0 executable

### ISSUE THREAD

**Issue by laanwj** [bitcoin/bitcoin#7582]
Reported on IRC by fakesatoshi. This crash seems to happen randomly while clicking:

```
[16419.562321] bitcoin-qt[1892]: segfault at 0 ip 00007fa82a53da1c sp 00007fffc4391810 error 4 in bitcoin-qt[7fa829fcf000+1e2e000]
[17045.871471] bitcoin-qt[2061]: segfault at 0 ip 00007f3b4b694a1c sp 00007ffc0739bd50 error 4 in bitcoin-qt[7f3b4b126000+1e2e000]
[20753.833830] bitcoin-qt[2891]: segfault at 0 ip 00007f7bfbff1a1c sp 00007ffd5ffe2880 error 4 in bitcoin-qt[7f7bfba83000+1e2e000]
```

This is offset 0x56ea1c in the 64-bit bitcoin-qt executable. Addr2line (on a non-stripped version of the executable) gives us:

```
addr2line -aCf -e bitcoin-0.12.0/bin/bitcoin-qt 0x56ea1c
0x000000000056ea1c
QXcbScreen::mapFromNative(QPoint const&) const
:?
```

Operating system: Debian Jessie

```
Linux vbox 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt20-1+deb8u3 (2016-01-17) x86_64 GNU/Linux
```

No full traceback is available as it doesn't seem to be reproducible in gdb.

From at the assembly around the crash eip it looks like the first argument, the implicit `this` argument is 0.

```
000000000056ea10 <_ZNK10QXcbScreen13mapFromNativeERK6QPoint>:
  56ea10:       55                      push   %rbp
  56ea11:       48 89 f5                mov    %rsi,%rbp
  56ea14:       53                      push   %rbx
  56ea15:       48 89 fb                mov    %rdi,%rbx
  56ea18:       48 83 ec 08             sub    $0x8,%rsp
  56ea1c:       48 8b 07                mov    (%rdi),%rax
```

What could cause an XcbScreen to be 0? @theuni any idea?

**Comment by laanwj** [comment]
Wonder if this is related to #5910 again.

**Comment by laanwj** [comment]
Possibly related upstream issues with similar backtraces:
- https://bugreports.qt.io/browse/QTBUG-47586
- https://lists.debian.org/debian-qt-kde/2015/12/msg00355.html
- https://bugs.kde.org/show_bug.cgi?id=350566

**Comment by fanquake** [comment]
Reading through the linked bug reports, and some others [here](https://bugreports.qt.io/browse/QTBUG-42985), [here](https://bugreports.qt.io/browse/QTBUG-47033) and [here](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=807528) this should have been fixed in Qt 5.6.0.
